### ISSUE TITLE
Incorrectly flagging incompatible type for List of Tuple of Union

### ISSUE THREAD

**Issue by henridwyer** [python/mypy#3849]
Here's a minimal example that fails under mypy 0.521

``` python
from typing import List, Tuple, Union

dic = [(1, b'a')]


def fun(arg: List[Tuple[Union[str, int], bytes]]) -> None:
    print(arg)


fun(dic)
```

```
Argument 1 to "fun" has incompatible type List[Tuple[int, bytes]]; expected List[Tuple[Union[str, int], bytes]]
```

But this should work since List[Tuple[int, bytes]] is an instance of List[Tuple[Union[str, int], bytes]]

**Comment by ilevkivskyi** [comment]
Thanks for reporting! I think mypy is right here since ``List`` is invariant. Also on master we now have much clearer error message for this code snippet:
```
error: Argument 1 to "fun" has incompatible type List[Tuple[int, bytes]]; expected List[Tuple[Union[str, int], bytes]]
note: "List" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance
note: Consider using "Sequence" instead, which is covariant
```
